src/talker.h

```diff
@@ -0,0 +1,293 @@
+#pragma once
+#ifndef CATA_SRC_TALKER_H
+#define CATA_SRC_TALKER_H
+
+class faction;
+class item;
+class mission;
+class npc;
+class player;
+class recipe;
+struct tripoint;
+class vehicle;
+
+/*
+ * Talker is an entity independent way of providing a participant in a dialogue.
+ * Talker is a virtual abstract class and should never really be used.  Instead,
+ * entity specific talker child classes such as character_talker should be used.
+ */
+class talker
```

Comment:
You've provided default implementations for all of these functions.  I'm not convinced all of them make sense.  Did you consider have pure virtual functions instead (with no implementation)?

Reply:
My logic was that I want any child talker class to only need to implement the parts of the API it actually cares about, and using a dialogue condition or effect that uses an unspecified child function should not cause a run time error but shouldn't necessarily to do anything.

ie, a `talker_item` is probably not going to have a meaningful implementation of `get_skill_level()`, but a mod developer might invoke `npc_has_skill_level` in JSON, and I don't want that to cause an error.

Doing it this way has already bit me in testing (I had forgotten to write `talker_character::has_trait_flag()` and that caused some tests to fail) but I'd rather that tests fail then the runtime does whatever weird thing C++ does when you have an unimplemented pure virtual function - or alternately, forcing `talker_item`, `talker_monster`, `talker_console`, etc to each independently reimplement `num_bionics()`.

Reply:
Making them pure virtual would force every derived class to implement them.

Some are reasonable (like `num_bionics` defaulting to zero), but others seem risky, like returning `point_zero` for location.  I can see that causing weird behaviour.

Reply:
Good point.
I'm making the pos*() functions purely virtual, and removing global_omt_goal().  The other functions have reasonable default values.